src/tallies/trigger.cpp

```diff
@@ -0,0 +1,203 @@
+#include "openmc/tallies/trigger.h"
+
+#include <cmath>
+#include <sstream>
+#include <utility> // for std::pair
+
+#include "openmc/capi.h"
+#include "openmc/constants.h"
+#include "openmc/error.h"
+#include "openmc/reaction.h"
+#include "openmc/settings.h"
+#include "openmc/simulation.h"
+#include "openmc/tallies/tally.h"
+
+namespace openmc {
+
+//==============================================================================
+// Global variable definitions
+//==============================================================================
+
+namespace settings {
+  KTrigger keff_trigger;
+}
+
+//==============================================================================
+// Non-member functions
+//==============================================================================
+
+std::pair<double, double>
+get_tally_uncertainty(int i_tally, int score_index, int filter_index)
+{
+  const auto& tally {model::tallies[i_tally]};
+
+  auto sum = tally->results_(filter_index, score_index, RESULT_SUM);
+  auto sum_sq = tally->results_(filter_index, score_index, RESULT_SUM_SQ);
+
+  int n = tally->n_realizations_;
+  auto mean = sum / n;
+  double std_dev = std::sqrt((sum_sq/n - mean*mean) / (n-1));
+  double rel_err = (mean != 0.) ? std_dev / std::abs(mean) : 0.;
+
+  return {std_dev, rel_err};
+}
+
+//! Find the limiting limiting tally trigger.
+//
+//! param[out] ratio The uncertainty/threshold ratio for the most limiting
+//!   tally trigger
+//! param[out] tally_id The ID number of the most limiting tally
+//! param[out] score The most limiting tally score bin
+
```

Comment:
It seems the style elsewhere is to have functions take pointers rather than references. We should probably be consistent.